### PR TITLE
test(client): sign in

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

* test(client): sign in
* add `-gcflags=all=-l` in test command, for the mock util need to [disable inlining](https://github.com/agiledragon/gomonkey#notes).

## Testing

UT

### CODE REVIEW

File: client/sign_in_test.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	})
+	defer patchesForGenerateCodeVerifier.Reset()
+
+	patchesForGenerateCodeChallenge := gomonkey.ApplyFunc(core.GenerateCodeChallenge, func(codeVerifier string) string {
+		return testCodeChallenge
+	})
+	defer patchesForGenerateCodeChallenge.Reset()
+
+	patchesForGenerateSignInUri := gomonkey.ApplyFunc(core.GenerateSignInUri, func(option *core.SignInUriGenerationOptions) (string, error) {
+		return testAuthEndpoint + "?redirect_uri=" + testRedirectUri + "&state=" + testState + "&code_challenge=" + testCodeChallenge, nil
```
Comment (wangsijie) [inline client/sign_in_test.go:37]: can we just return "generated_sign_in_uri" here? for unit test, we don't need the real logic.

File: client/sign_in_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	})
+	defer patchesForGenerateSignInUri.Reset()
+
+	var logtoClientSpy *LogtoClient
+
+	patchesForFetchOidcConfig := gomonkey.ApplyPrivateMethod(logtoClientSpy, "fetchOidcConfig", func(_ *LogtoClient) (core.OidcConfigResponse, error) {
```
Comment (wangsijie) [inline client/sign_in_test.go:43]: nit: can be moved the the front?

File: client/sign_in_test.go (hunk 1 above)
Comment (xiaoyijun) [reply client/sign_in_test.go]: cool, updated.

File: client/sign_in_test.go (hunk 2 above)
Comment (xiaoyijun) [reply client/sign_in_test.go]: updated.

**Review by wangsijie — APPROVED** [review]
LGTM

**Pull request merged**